setup

```diff
@@ -142,6 +137,11 @@ while [[ $# -gt 0 ]]; do
 done
 
 # Figure out the options, execute setup
+echo "setup has started..."
```

Comment:
Ok you did shift these. can you merge this one with the getopt PR?

Reply:
sure